src/modules.js

```diff
@@ -84,27 +84,33 @@ var LibraryManager = {
       // Core filesystem libraries (always linked against, unless -s FILESYSTEM=0 is specified)
       libraries = libraries.concat([
         'library_fs.js',
-        'library_memfs.js',
         'library_tty.js',
-        'library_pipefs.js',
+        'library_pipefs.js', // ok to include it by default since it's only used if the syscall is used
       ]);
 
+      if (MEMFS) {
+        libraries.push('library_memfs.js');
+      }
+
       // Additional filesystem libraries (without AUTO_JS_LIBRARIES, link to these explicitly via -lxxx.js)
       if (AUTO_JS_LIBRARIES) {
         if (ENVIRONMENT_MAY_BE_WEB || ENVIRONMENT_MAY_BE_WORKER) {
-          libraries = libraries.concat([
-            'library_idbfs.js',
-            'library_proxyfs.js',
-            'library_sockfs.js',
-            'library_workerfs.js',
-          ]);
+          if (IDBFS) {
+            libraries.push('library_idbfs.js');
+          }
+          if (PROXYFS) {
+            libraries.push('library_proxyfs.js');
+          }
+          if (WORKERFS) {
+            libraries.push('library_workerfs.js');
+          }
```

Comment:
Now that these are controlled explicitly I think we can/should move them outside of `AUTO_JS_LIBRARIES` no?  Since they are no longer "auto" as such.

Reply:
Actually not that I think about it why not ask people to use the normally `-lproxyfs` syntax which we support for js libraries rather than adding these new settings?

Reply:
Ah, good idea, I rewrote it that way, PTAL.

This way we can't disable MEMFS, as it's on by default, but that seems a minor issue, and we're using a more conventional linking mechanism.